### PR TITLE
[#IOCIT-137] Add pushNotificationsContentType optional field to the profile model

### PR DESCRIPTION
**Pull request by arcogabbo**
#### List of Changes

* added `pushNotificationsContentType` optional field to the profile model
* added `push_notifications_content_type` optional field to the OpenAPI spec(in particular to the `Profile` and `ExtendedProfile` definitions)

#### Motivation and Context

allow user to specify how to visualize the push notifications

#### How Has This Been Tested?

build passed 
unit and integration tests passed

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by michaeldisaro — APPROVED** [review]
LGTM

File: src/models/profile.ts [hunk 1]
```diff
@@ -117,6 +118,14 @@ export const Profile = t.intersection([
     // https://it.wikipedia.org/wiki/ISO_3166-2
     preferredLanguages: PreferredLanguages,
 
+    // This parameter specifies how a specific user wants to visualize push notifications.
+    // FULL leads to descriptive push notifications while ANONYMOUS leads to silent ones.
+    // (defaults to UNSET)
+    pushNotificationsContentType: withDefault(
+      t.union([PushNotificationsContentType, t.literal("UNSET")]),
+      "UNSET"
```
Comment (BurnedMarshal) [inline src/models/profile.ts:126]: What if we define a global `UNSET` type for this scenario?
```typescrypt
export const UNSET = t.literal("UNSET");
export type UNSET = t.TypeOf<typeof UNSET>;
export const UNSET_VALUE: UNSET = "UNSET";
```

```suggestion
      t.union([PushNotificationsContentType, UNSET]),
      UNSET_VALUE
```

File: src/models/__tests__/profile.test.ts [hunk 2]
```diff
@@ -112,6 +112,50 @@ describe("findLastVersionByModelId", () => {
     }
   );
 
+  it.each`
```
Comment (BurnedMarshal) [inline src/models/__tests__/profile.test.ts:115]: We should add even a test for the other model methods like create and upsert.

File: src/models/__tests__/profile.test.ts [hunk 3]
```diff
@@ -112,6 +112,50 @@ describe("findLastVersionByModelId", () => {
...
+    case                                                              | pushNotificationsContentType | expectedPushNotificationsContentType
+    ${"existing profile with pushNotificationsContentType empty"}     | ${undefined}                 | ${"UNSET"}
```
Comment (BurnedMarshal) [inline src/models/__tests__/profile.test.ts:117]: Add `UNSET` -> `UNSET` scenario

File: __integrations__/models/profile.test.ts [hunk 4]
```diff
@@ -478,4 +479,41 @@ describe("Models |> Profile", () => {
       context.dispose();
     }
   );
+
+  it.each`
+    inputValue     | expectedValue
+    ${undefined}   | ${"UNSET"}
```
Comment (BurnedMarshal) [inline __integrations__/models/profile.test.ts:485]: ditto

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: __integrations__/models/profile.test.ts (hunk 4 above)
Comment (arcogabbo) [reply __integrations__/models/profile.test.ts:485]: done in https://github.com/pagopa/io-functions-commons/pull/297/commits/25f802f12c34ec047393044f870ae527936bc452

File: src/models/__tests__/profile.test.ts (hunk 3 above)
Comment (arcogabbo) [reply src/models/__tests__/profile.test.ts:126]: done in https://github.com/pagopa/io-functions-commons/pull/297/commits/25f802f12c34ec047393044f870ae527936bc452

File: src/models/profile.ts (hunk 1 above)
Comment (arcogabbo) [reply src/models/profile.ts]: done in https://github.com/pagopa/io-functions-commons/pull/297/commits/25f802f12c34ec047393044f870ae527936bc452

File: src/models/__tests__/profile.test.ts (hunk 2 above)
Comment (arcogabbo) [reply src/models/__tests__/profile.test.ts:124]: done in https://github.com/pagopa/io-functions-commons/pull/297/commits/e0d12ae28c6416710e4e67905d9ef48523160e9e

File: src/models/__tests__/profile.test.ts [hunk 5]
```diff
@@ -275,6 +319,51 @@ describe("createProfile", () => {
...
+      const model = new ProfileModel(containerMock);
+
+      const newProfileWithNotificationsContentType: NewProfile = {
+        ...newProfile,
+        pushNotificationsContentType
+      };
+      const result = await model.create(
+        newProfileWithNotificationsContentType
+      )();
+
```
Comment (BurnedMarshal) [inline src/models/__tests__/profile.test.ts:357]: Check that we are calling the `create` method with the right parameters, if we don't we are testing the mock result only.
```suggestion
      expect(containerMock.items.create).toBeCalledTimes(1);
      expect(containerMock.items.create).toHaveBeenCalledWith(
        expect.objectContaining({
          kind: newProfileWithNotificationsContentType.kind,
          fiscalCode: newProfileWithNotificationsContentType.fiscalCode,
          pushNotificationsContentType: expectedPushNotificationsContentType,
          id: generateVersionedModelId<Profile, "fiscalCode">(
            newProfileWithNotificationsContentType.fiscalCode,
            0 as NonNegativeInteger
          ),
          version: 0
        }),
        expect.anything()
      );
```

**Review by BurnedMarshal — APPROVED** [review]
LGTM apart a minor unit test improvement

File: src/models/__tests__/profile.test.ts [hunk 6]
```diff
@@ -355,6 +356,25 @@ describe("createProfile", () => {
...
+            newProfileWithNotificationsContentType.fiscalCode,
+            0 as NonNegativeInteger
+          ),
+          version: 0,
+          // the decoder here sets the field to UNSET for undefined values
+          pushNotificationsContentType:
+            newProfileWithNotificationsContentType.pushNotificationsContentType !==
+            undefined
+              ? newProfileWithNotificationsContentType.pushNotificationsContentType
+              : "UNSET"
```
Comment (BurnedMarshal) [inline src/models/__tests__/profile.test.ts:374]: Why this login in here? IMHO is the same of `expectedPushNotificationsContentType`

Comment (arcogabbo) [reply src/models/__tests__/profile.test.ts]: yes it is unnecessary as you said, sorry.

**Pull request merged**